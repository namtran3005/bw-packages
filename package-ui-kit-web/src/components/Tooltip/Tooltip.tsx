import React from 'react';
import { Tooltip as MuiTooltip } from '@material-ui/core';
import { TooltipProps as MuiTooltipProps } from '@material-ui/core/Tooltip';

import { Typography } from '../Typography';

const Tooltip = React.forwardRef<HTMLDivElement, MuiTooltipProps>(
  ({ title, ...cleanProps }, ref) => {
    return (
      <MuiTooltip
        ref={ref}
        title={
          <Typography.Caption foreground="white">{title}</Typography.Caption>
        }
        {...cleanProps}
      />
    );
  }
);

export { Tooltip, MuiTooltipProps as TooltipProps };
