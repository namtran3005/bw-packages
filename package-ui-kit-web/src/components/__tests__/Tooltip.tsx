import * as React from 'react';
import { shallow } from 'enzyme';
import { Tooltip } from '../Tooltip';

const defaultProps = {
  title: 'Tooltip content',
  children: <div />,
};

describe('Tooltip', () => {
  it('should render tooltip', () => {
    const wrapper = shallow(<Tooltip {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
