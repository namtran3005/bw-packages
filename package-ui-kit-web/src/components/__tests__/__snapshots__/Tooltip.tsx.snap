// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render tooltip 1`] = `
<WithStyles(ForwardRef(Tooltip))
  title={
    <Caption
      foreground="white"
    >
      Tooltip content
    </Caption>
  }
>
  <div />
</WithStyles(ForwardRef(Tooltip))>
`;
